db/seeds/source_types.rb

```diff
@@ -8,15 +8,25 @@ def update_or_create(attributes)
 end
 
 openshift_json_schema = {
-  :title  => "Configure OpenShift",
-  :fields => [
-    {:component => "text-field", :name => "endpoint.role", :type => "hidden", :initialValue => "kubernetes"},
-    {:component => "text-field", :name => "authentication.authtype", :type => "hidden", :initialValue => "token"},
-    {:component => "text-field", :name => "url", :label => "URL", :validate => [{:type => "url-validator"}]},
-    {:component => "checkbox", :name => "endpoint.verify_ssl", :label => "Verify SSL"},
-    {:component => "text-field", :name => "endpoint.certificate_authority", :label => "Certificate Authority", :condition => {:when => "endpoint.verify_ssl", :is => true}},
-    {:component => "text-field", :name => "authentication.password", :label => "Token", :type => "password"}
-  ]
+  :authentication => [{
+    :type   => 'token',
+    :meta   => {
```

Comment:
What else might be in this `:meta` section?  I'm with @slemrmartin any reason not just to move `:name` (or `:label` like the `:fields`)

Reply:
I initially thought that there would be some description.

I think `:name` is good choice.